libsolidity/ast/Types.cpp

```diff
@@ -3090,6 +3090,54 @@ string FunctionType::canonicalName() const
 	return "function";
 }
 
+string FunctionType::humanReadableName() const
+{
+	string name;
+	switch (m_kind)
+	{
+		case Kind::Error:
+			name = "error " + m_declaration->name();
+			name += '(';
+			for (auto it = m_parameterTypes.begin(); it != m_parameterTypes.end(); ++it)
+				name += (*it)->toString(true) + (it + 1 == m_parameterTypes.end() ? "" : ",");
+			name += ")";
+			if (m_stateMutability != StateMutability::NonPayable)
+				name += " " + stateMutabilityToString(m_stateMutability);
+			if (m_kind == Kind::External)
+				name += " external";
+			if (!m_returnParameterTypes.empty())
+			{
+				name += " returns (";
+				for (auto it = m_returnParameterTypes.begin(); it != m_returnParameterTypes.end(); ++it)
+					name += (*it)->toString(true) + (it + 1 == m_returnParameterTypes.end() ? "" : ",");
+				name += ")";
+			}
+			break;
+		case Kind::Event:
+			name = "event " + m_declaration->name();
+			name += '(';
```

Comment:
Should we refactor out the logic for stringifying parameter types? Now it's duplicated here and in `FunctionType::toString`?